lib/improver/nbhood/use_nbhood.py

```diff
@@ -170,4 +212,205 @@ def process(self, cube, mask_cube):
+        easily done by renormalization of the weights.
+
+        Once we have valid weights for adjacent bands for each point we can
+        collapse the "topographic_zone" coordinate using a weighted mean.
+
+        When this plugin is used alongside :class:`~improver.nbhood.use_nbhood.\
+ApplyNeighbourhoodProcessingWithAMask` we end up with a cube with the same
+        dimensions as the original cube, but the neighbourhood processing
+        has been applied using masks so that only similar points are used in
+        the neighbourhood.
+
+    """
+
+    def __init__(self, coord_masked, weights):
+        """
+        Initialise the class.
+
+        Args:
+            coord_masked (string):
+                String matching the name of the coordinate that has been used
+                for masking.
+            weights (cube):
+                A cube from an ancillary file containing the weights for each
+                point in the coord_masked at each grid point. Only two points
+                in coord_masked can have a non-zero weight for each grid-point,
+                i.e. we are only weighting between two adjacent bands in the
+                neighbourhood output for each gridpoint.
+                Should have the coordinates coord_masked, x and y.
+                The weights cube can be masked, and this mask will be retained,
+                and will be present in the output.
+
+        """
+        self.coord_masked = coord_masked
+        self.weights = weights
+
+    def __repr__(self):
+        """Represent the configured plugin instance as a string."""
+        result = ('<ApplyNeighbourhoodProcessingWithAMask: '
+                  'coord_masked: {}, weights: {}>')
+        return result.format(self.coord_masked,
+                             self.weights)
+
+    def renormalize_weights(self, nbhood_cube):
+        """
+        Renormalize the weights taking into account where there are NaNs in the
+        result from neighbourhood.
+
+        The weights corresponding to NaNs in the result from neighbourhooding
+        with a mask are set to zero and then the weights are renormalized along
+        the axis corresponding to the coordinate we want to collapse.
+
+        Args:
+            nbhood_cube (iris.cube.Cube):
+                The cube that has been through masked neighbourhood processing
+                and has the dimension we wish to collapse. Must have the same
+                dimensions of the cube.
+        """
+        # If the weights are masked we want to retain the mask.
+        condition = np.isnan(nbhood_cube.data)
+        if ma.is_masked(self.weights.data):
+            condition = condition & ~self.weights.data.mask
+
+        self.weights.data[condition] = 0.0
+        axis = nbhood_cube.coord_dims(self.coord_masked)
+        self.weights.data = WeightsUtilities.normalise_weights(
+            self.weights.data, axis=axis)
+
+    def fix_metadata(self, result_cube):
```

Comment:
More descriptive name?  How about "remove_collapsed_coordinate" or "remove_collapsed_coord_refs()"?

Reply:
Changed to remove_collapsed_coord_refs